de-lib/src/main/java/org/iplantc/de/apps/client/views/tools/EditToolViewImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.iplantc.de.apps.client.views.tools;
+
+import org.iplantc.de.client.models.tool.Tool;
+import org.iplantc.de.client.models.tool.ToolAutoBeanFactory;
+import org.iplantc.de.client.models.tool.ToolContainer;
+import org.iplantc.de.client.models.tool.ToolImage;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.uibinder.client.UiTemplate;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import com.google.web.bindery.autobean.shared.AutoBeanCodex;
+import com.google.web.bindery.autobean.shared.AutoBeanUtils;
+
+import com.sencha.gxt.widget.core.client.FramedPanel;
+import com.sencha.gxt.widget.core.client.container.VBoxLayoutContainer;
+import com.sencha.gxt.widget.core.client.form.FormPanelHelper;
+import com.sencha.gxt.widget.core.client.form.IsField;
+import com.sencha.gxt.widget.core.client.form.TextArea;
+import com.sencha.gxt.widget.core.client.form.TextField;
+
+import java.util.List;
+
+/**
+ * Created by sriram on 4/27/17.
+ */
+public class EditToolViewImpl implements EditToolView {
+
+    @Inject
+    ToolAutoBeanFactory factory;
+
+    @UiField
+    VBoxLayoutContainer container;
+
+    @UiField
+    FramedPanel form;
+
+    @UiField
+    TextField name;
+
+    @UiField
+    TextArea desc;
+
+    @UiField
+    TextField  imgName;
+
+    @UiField
+    TextField tag;
+
+    @UiField
+    TextField url;
+
+    @Override
+    public Widget asWidget() {
+        return container;
+    }
+
+    @Override
+    public boolean validate() {
+        boolean valid = true;
+        List<IsField<?>> fields = FormPanelHelper.getFields(form);
+        for (IsField<?> f : fields) {
+            if (!f.isValid(false)) {
+                GWT.log(f.toString() + "-> " + f.getValue());
+                valid = false;
+                break;
+            }
+        }
+
+        return valid;
+    }
+
+    @UiTemplate("EditToolView.ui.xml")
+    interface EditToolViewUiBinder extends UiBinder<Widget, EditToolViewImpl> {
+        
+    }
+
+    private static  final  EditToolViewUiBinder uiBinder = GWT.create(EditToolViewUiBinder.class);
+
+    @Inject
+    public EditToolViewImpl() {
+        uiBinder.createAndBindUi(this);
+    }
+
+    @Override
+    public Tool getTool() {
+        Tool tool = AutoBeanCodex.decode(factory, Tool.class,"{}").as();
+        tool.setName(name.getValue());
+        tool.setDescription(desc.getValue());
+        tool.setLocation(url.getValue());
+
+        ToolImage image = AutoBeanCodex.decode(factory, ToolImage.class, "{}").as();
+        image.setName(imgName.getValue());
+        image.setTag(tag.getValue());
+        image.setUrl(url.getValue());
+
+        ToolContainer container = AutoBeanCodex.decode(factory, ToolContainer.class, "{}").as();
+        container.setImage(image);
+        tool.setContainer(container);
```

Comment:
This could be a preference thing, but I found it useful to use `Editor` instead of all these get/set operations, like in `ToolAdminDetailsView` and all the other sub-Editors in `org.iplantc.de.admin.desktop.client.toolAdmin.view.subviews`.  I'm not sure, but you might even be able to reuse some of those?